tap_salesforce/__init__.py

```diff
@@ -268,8 +338,32 @@ def do_discover(sf):
     entries = []
 
     # Check if the user has BULK API enabled
-    if sf.api_type == 'BULK' and not Bulk(sf).has_permissions():
-        raise TapSalesforceBulkAPIDisabledException('This client does not have Bulk API permissions, received "API_DISABLED_FOR_ORG" error code')
+    if sf.api_type == "BULK" and not Bulk(sf).has_permissions():
+        raise TapSalesforceBulkAPIDisabledException(
+            'This client does not have Bulk API permissions, received "API_DISABLED_FOR_ORG" error code'
+        )
+
+    objects_list = sorted(objects_to_discover)
+    start_counter = 0
+    concurrency_limit = 25
+
+    while start_counter < len(objects_list):
+        end_counter = start_counter + concurrency_limit
+        if end_counter >= len(objects_list):
+            end_counter = len(objects_list)
+
+        chunk = objects_list[start_counter:end_counter]
+        chunk_args = [
+            [
+                sf,
+                sobject_name,
+                entries,
+                sf_custom_setting_objects,
+                object_to_tag_references,
+            ]
+            for sobject_name in chunk]
+        run_concurrently(discover_stream, chunk_args)
+        start_counter = end_counter
 
     for sobject_name in sorted(objects_to_discover):
```

Comment:
Shouldn't we get rid of the for loop now since the work is being done above?

Reply:
old loop deleted